systemtest/src/test/java/io/strimzi/systemtest/upgrade/ZookeeperUpgradeST.java

```diff
@@ -0,0 +1,187 @@
+                .editSpec()
+                    .editKafka()
+                        .withVersion(initialVersion.version())
+                        .addToConfig("log.message.format.version", logMsgFormat)
+                    .endKafka()
+                .endSpec()
+                .done();
+
+        LOGGER.info("Deployment of initial Kafka version (" + initialVersion.version() + ") complete");
+
+        String zkVersionCommand = "ls libs | grep -Po 'zookeeper-\\K\\d+.\\d+.\\d+' | head -1";
+        String zkResult = cmdKubeClient().execInPodContainer(KafkaResources.zookeeperPodName(CLUSTER_NAME, 0),
+                "zookeeper", "/bin/bash", "-c", zkVersionCommand).out().trim();
+        LOGGER.info("Pre-change Zookeeper version query returned: " + zkResult);
+
+        String kafkaVersionCommand = "ls libs | grep -Po 'kafka_\\d+.\\d+-\\K(\\d+.\\d+.\\d+)(?=.jar)' | head -1";
+        String kafkaResult = cmdKubeClient().execInPodContainer(KafkaResources.kafkaPodName(CLUSTER_NAME, 0),
+                "kafka", "/bin/bash", "-c", kafkaVersionCommand).out().trim();
+        LOGGER.info("Pre-change Kafka version query returned: " + kafkaResult);
+
+        Map<String, String> zkPods = StatefulSetUtils.ssSnapshot(KafkaResources.zookeeperStatefulSetName(CLUSTER_NAME));
+        Map<String, String> kafkaPods = StatefulSetUtils.ssSnapshot(KafkaResources.kafkaStatefulSetName(CLUSTER_NAME));
+
+        LOGGER.info("Updating Kafka CR version field to " + newVersion.version());
+
+        // Get the Kafka resource from K8s
+        KafkaResource.replaceKafkaResource(CLUSTER_NAME, kafka -> {
+            kafka.getSpec().getKafka().setVersion(newVersion.version());
+        });
+
+        Kafka retrievedKafka = Crds.kafkaOperation(kubeClient(NAMESPACE).getClient())
+                .inNamespace(NAMESPACE)
+                .withName(CLUSTER_NAME)
+                .get();
+
+        // Change the Kafka version for the resource
+        retrievedKafka.getSpec().getKafka().setVersion(newVersion.version());
+
+        // Patch the existing resource with this new version
+        Crds.kafkaOperation(kubeClient().getClient()).inNamespace(NAMESPACE).withName(CLUSTER_NAME).patch(retrievedKafka);
+
+        LOGGER.info("Waiting for deployment of new Kafka version (" + newVersion.version() + ") to complete");
+
+        // Wait for the zk version change roll
+        zkPods = StatefulSetUtils.waitTillSsHasRolled(KafkaResources.zookeeperStatefulSetName(CLUSTER_NAME), zkReplicas, zkPods);
+        LOGGER.info("1st Zookeeper roll (image change) is complete");
+
+        // Wait for the kafka broker version change roll
+        kafkaPods = StatefulSetUtils.waitTillSsHasRolled(KafkaResources.kafkaStatefulSetName(CLUSTER_NAME), kafkaReplicas, kafkaPods);
+        LOGGER.info("Kafka roll (image change) is complete");
+
+        PodUtils.waitUntilPodReplicasCount(KafkaResources.kafkaStatefulSetName(CLUSTER_NAME), 2);
```

Comment:
You run with 3 replicas, so you should wait for 3 pods (the same below as well).